src/Databases/Mongo/src/MongoDBAtlasVectorCollection.cs

```diff
@@ -0,0 +1,75 @@
+using LangChain.Databases.Mongo.Client;
+using MongoDB.Bson.Serialization;
+using MongoDB.Driver;
+
+namespace LangChain.Databases.Mongo;
+
+public class MongoDBAtlasVectorCollection(
+    IMongoContext mongoContext,
+    string indexName,
+    string name = VectorCollection.DefaultName,
+    string? id = null)
+: VectorCollection(name, id), IVectorCollection
+{
+    protected IMongoContext MongoContext { get; } = mongoContext;
+    protected IMongoCollection<Vector> mongoCollection => MongoContext.GetCollection<Vector>(Name);
```

Comment:
> Consider caching the `mongoCollection` to avoid repeated database calls.

<!-- This is an auto-generated comment by CodeRabbit -->

Comment from a bot, but perhaps it’s worth moving IMongoCollection<Vector> to the constructor? And the Database implementation will contain the IMongoContext itself

Reply:
I have assigned the collection into a private member at time of constructor. Also remove the handle to the MongoContext as it is no longer need once constructed.